src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -114,6 +109,11 @@ public void Start()
 
             logger.Info($"Stopping notifier for '{fullPath}'");
 
+            foreach (var option in options)
+            {
+                option.ExceptionReceived -= OptionsOnExceptionReceived;
+            }
+
             var timeoutTask = Task.Delay(TimeSpan.FromSeconds(30));
             var allTasks = pipelineInvocationTasks.Values;
             var finishedTask = await Task.WhenAny(Task.WhenAll(allTasks), timeoutTask).ConfigureAwait(false);
```

Comment:
I'm good with the current approach. I'll create the separate issue to address that.